overlord/storestate/storestate.go

```diff
@@ -0,0 +1,163 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package storestate
+
+import (
+	"fmt"
+	"net/url"
+
+	"golang.org/x/net/context"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/progress"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/store"
+)
+
+var storeNew = store.New
+
+// StoreState holds the state for the store in the system.
+type StoreState struct {
+	// StoreLocation is the store API's base location.
+	StoreLocation string `json:"store-location"`
```

Comment:
s/store-location/base-url/ ?

Reply:
I went with the "Store" prefix because `state.Location` didn't make sense when reading it aloud- it's not the state's location - and that was one of the complaints in the earlier review.

"Location" was a suggestion in the review and seemed nicer to read than "URL". Personally, I'm not sure "base" adds anything other than "BaseURL" being a bit nicer than plain "URL".

Anyway ... I will take your guidance on the best name and then change it to whatever is recommended. Just let me know.

Reply:
I don't understand the comment, it's inside "store" in state,  and StoreState has already Store in its name.

storestate.StoreState.StoreLocation

has a bit too many "store" in it.

I'm not a fan of location here  because it seems user oriented,  this is the api-side base url instead.

Also store the assertion has a url and location, this is the url not the location

Reply:
to be super clear, my starting point  is that we have:

* store.url
* and cfg.SetAPI  which I think maybe is better called cfg.SetBaseURL

and here they need to meet somehow

Reply:
Changed to BaseURL/base-url.